curator-framework/src/main/java/org/apache/curator/framework/imps/NamespaceWatcher.java

```diff
@@ -75,4 +89,53 @@ else if ( curatorWatcher != null )
             }
         }
     }
+
+    // specialized equals()/hashCode() that makes this instance equal to the actual watcher
+
+    @Override
+    public boolean equals(Object o)
+    {
+        if ( this == o )
+        {
+            return true;
+        }
+        if ( o == null )
+        {
+            return false;
+        }
+
+        if ( getClass() == o.getClass() )
```

Comment:
This is IntelliJ's auto-gen again the entire equals()/hashCode()/toString() methods are auto generated.